x-pack/plugin/runtime-fields/src/test/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptFieldTypeTestCase.java

```diff
@@ -6,29 +6,173 @@
 import java.util.function.BiConsumer;
 
+import static java.util.stream.Collectors.toList;
 import static org.hamcrest.Matchers.equalTo;
 import static org.mockito.Matchers.anyString;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-abstract class AbstractScriptFieldTypeTestCase extends ESTestCase {
+public abstract class AbstractScriptFieldTypeTestCase extends ESTestCase {
+
+    private static final ToXContent.Params INCLUDE_DEFAULTS = new ToXContent.MapParams(Map.of("include_defaults", "true"));
+
     protected abstract MappedFieldType simpleMappedFieldType() throws IOException;
 
     protected abstract MappedFieldType loopFieldType() throws IOException;
 
-    protected abstract String runtimeType();
+    protected abstract String typeName();
+
+    public final void testMinimalSerializesToItself() throws IOException {
+        XContentBuilder orig = JsonXContent.contentBuilder().startObject();
+        createMapperService(runtimeFieldMapping(this::minimalMapping)).documentMapper().mapping().toXContent(orig, ToXContent.EMPTY_PARAMS);
+        orig.endObject();
+        XContentBuilder parsedFromOrig = JsonXContent.contentBuilder().startObject();
+        createMapperService(orig).documentMapper().mapping().toXContent(parsedFromOrig, ToXContent.EMPTY_PARAMS);
+        parsedFromOrig.endObject();
+        assertEquals(Strings.toString(orig), Strings.toString(parsedFromOrig));
+    }
+
+    public final void testMeta() throws IOException {
+        XContentBuilder mapping = runtimeFieldMapping(b -> {
+            minimalMapping(b);
+            b.field("meta", Collections.singletonMap("foo", "bar"));
+        });
+        MapperService mapperService = createMapperService(mapping);
+        assertEquals(
+            XContentHelper.convertToMap(BytesReference.bytes(mapping), false, mapping.contentType()).v2(),
+            XContentHelper.convertToMap(mapperService.documentMapper().mappingSource().uncompressed(), false, mapping.contentType()).v2()
+        );
+
+        mapping = runtimeFieldMapping(this::minimalMapping);
+        merge(mapperService, mapping);
+        assertEquals(
+            XContentHelper.convertToMap(BytesReference.bytes(mapping), false, mapping.contentType()).v2(),
+            XContentHelper.convertToMap(mapperService.documentMapper().mappingSource().uncompressed(), false, mapping.contentType()).v2()
+        );
+
+        mapping = runtimeFieldMapping(b -> {
+            minimalMapping(b);
+            b.field("meta", Collections.singletonMap("baz", "quux"));
+        });
+        merge(mapperService, mapping);
+        assertEquals(
+            XContentHelper.convertToMap(BytesReference.bytes(mapping), false, mapping.contentType()).v2(),
+            XContentHelper.convertToMap(mapperService.documentMapper().mappingSource().uncompressed(), false, mapping.contentType()).v2()
+        );
+    }
+
+    public final void testMinimalMappingToMaximal() throws IOException {
```

Comment:
these three tests I borrowed from MapperTestCase , they were previously run as part of RuntimeFieldMapperTests